### PR TITLE
Milestone 9: Authorization

### PR DESCRIPTION
**Pull request by eroiyo**
Hello Microverse Community, this is the milestone number nine, the next features have been implemented...

- [ ] Delete functions for Post and comments,

- [ ] The post delete function is available in the post shown

- [ ] The Comment delete function is available in all the comment lists

- [ ] Super User implemented

To test the project correctly, you will have to migrate all the information using rails db:migrate and fill the database using db:seed also, you may want to migrate the information in your testing environment

you will have to create a .env file in the root of the project with the next information inside to fill the enviorement variables

GMAIL_USERNAME=your_email@example.com
GMAIL_PASSWORD=password

if you want to test the administrator features you will have to connect with the super user

username = Admin@example.com
password = administrator

that's all!
thanks for reading

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -37,4 +37,17 @@ def create
...
+    poster_id = post.user_id
+    if can? :destroy, post
+      post.comments.each do |comment|
+        Comment.delete_by(id: comment.id)
+      end
+      Post.delete_by(id: post_id)
+    end
+    redirect_to("/users/#{poster_id}")
+  end
 end
```
Comment (AshakaE) [inline app/controllers/posts_controller.rb:53]: - [ ] Kindly add `CanCanCan` for Post authorization in Post Controller

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @eroiyo  🙋‍♀️, 

Great job so far! 👌
Pull request is 👍 
Git flow ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._

###  Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (eroiyo) [reply app/controllers/posts_controller.rb:53]: Hello, thanks for the reply!

currently, i am using a Cancancan library function there

that can function in the line 45 looks if the current user have the authorization

that is in the model generated by cancancan named, abilities.rb

Comment (AshakaE) [reply app/controllers/posts_controller.rb:53]: I believe you understood the assignment, the difference is we want the post controller to have already loaded and authorized the resource so you do not have to check with `can?` method for every action. see [reference](https://github.com/CanCanCommunity/cancancan#controller-helpers)

**Comment by AshakaE** [comment]
### Hi @eroiyo  🙋‍♀️, 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by eroiyo** [comment]
Thanks! i will merge it

**Pull request merged**